addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -8008,6 +8015,75 @@ registry.VersionControl = SnippetOptionWidget.extend({
         });
         return this._super(...arguments);
     },
+
+    //--------------------------------------------------------------------------
+    // Public
+    //--------------------------------------------------------------------------
+
+    /**
+     * @override
+     */
+    async updateUI() {
+        // Waiting for the options to be visible so the alert can be correctly
+        // displayed.
+        this.options.wysiwyg.waitForEmptyMutexAction().then(() => {
+            const alertEl = this.$el[0].querySelector("we-alert");
+            const optionSectionEl = this.$el[0].parentElement;
+            if (!alertEl || optionSectionEl.classList.contains("o_we_outdated_block_options")) {
+                return;
+            }
+            // Adding a class to hide the other options, to only have the alert
+            // displayed (so it is placed correctly).
+            optionSectionEl.classList.add("o_we_outdated_block_options");
+        });
+        return this._super(...arguments);
+    },
+
+    //--------------------------------------------------------------------------
+    // Handlers
+    //--------------------------------------------------------------------------
+
+    /**
+     * Called to replace an outdated snippet by its new version.
+     *
+     * @param {Event} ev
+     */
+    async _onReplaceSnippetClick(ev) {
+        // Getting the new block version.
+        const snippet = this.$target[0].dataset.snippet;
+        const snippetTagName = this.$target[0].tagName.toLowerCase();
+        const customizePanelEl = ev.target.closest(".o_we_customize_panel");
+        const newBlockEl = customizePanelEl.parentElement
+            .querySelector(`${snippetTagName}[data-snippet=${snippet}]`).cloneNode(true);
+        // Replacing the block.
+        this.options.wysiwyg.odooEditor.historyPauseSteps();
+        this.$target[0].classList.add("d-none"); // Hiding the block to replace it smoothly.
+        this.$target[0].insertAdjacentElement("beforebegin", newBlockEl);
+        // Initializing the new block as if it was dropped.
+        await this.options.wysiwyg.snippetsMenu.callPostSnippetDrop($(newBlockEl));
+        await new Promise(resolve => {
+            this.trigger_up("remove_snippet",
+                {$snippet: this.$target, onSuccess: resolve, shouldRecordUndo: false}
+            );
+        });
+        this.options.wysiwyg.odooEditor.historyUnpauseSteps();
+        newBlockEl.classList.remove("oe_snippet_body");
+        this.options.wysiwyg.odooEditor.historyStep();
+    },
+    /**
+     * Called to still access the options of an outdated block, despite the
+     * warning.
+     *
+     * @param {Event} ev
+     */
+    _onAccessOptionsClick(ev) {
+        const alertEl = ev.target.parentElement;
+        const optionSectionEl = alertEl.closest(".o_we_outdated_block_options");
+        alertEl.remove();
```

Comment:
For me, yes, because the user surely won't remember which snippets are outdated every time he re-enters edit mode :slightly_smiling_face: 
And with the previous version of the outdated alert, it was always displayed so I think it should at least always be there when we open the edit mode :wink: